commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/FieldDef.java

```diff
@@ -49,12 +48,17 @@ public FieldDef(FieldDef rhs)
         this.fieldName = rhs.fieldName;
         this.fieldType = rhs.fieldType;
         this.typeName = rhs.typeName;
-        this.defs = rhs.defs;
+
+        this.defs = new FieldDef[rhs.defs.length];
+        for (int i = 0; i < rhs.defs.length; i++)
+        {
+            this.defs[i] = new FieldDef(rhs.defs[i]);
```

Comment:
Good question, I changed this to construct new FieldDefs here because I am going to potentially modify the SrcType later for KEYED_INTS. So I cannot share child FieldDefs any longer, as modifying the projected record definition would modify the original record definitions fields as well.

Reply:
is ther

> Good question, I changed this to construct new FieldDefs here because I am going to potentially modify the SrcType later for KEYED_INTS. So I cannot share child FieldDefs any longer, as modifying the projected record definition would modify the original record definitions fields as well.

Fair enough, are there any situations where we could avoid creating the new FieldDef? If so, is it worthwhile attempting to detect the condition in order to avoid the ctr call?